### PR TITLE
Add `Copy Error` button to the error dialog

### PR DESCRIPTION
**Pull request by armaldio**
Show a messageBox with a "copy to clipboard" button
If app is not ready, fall back to errorBox

I've also set "noLink" to true to prevent Windows from parsing "common" buttons and render them differently

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,18 +19,35 @@ module.exports = options => {
...
+
+			if (isReady) {
+				dialog.showMessageBox({
+					type: 'error',
+					buttons: ['Copy to clipboard', 'Ok'],
+					defaultId: 1,
+					title,
+					noLink: true,
+					message: cleanStack(stack)
+				}, btnIndex => {
```
Comment (sindresorhus) [inline index.js]: I don't think this should be async. You don't want the app to continue doing stuff while this message box is shown. Maybe the app decides to quit and closing this box it.

File: index.js [hunk 2]
```diff
@@ -17,18 +19,35 @@ module.exports = options => {
...
 		}
 
 		if (options.showDialog) {
 			const stack = err ? (err.stack || err.message || err || '[No error message]') : '[Undefined error]';
-			showErrorBox(title, cleanStack(stack));
+
+			if (isReady) {
+				dialog.showMessageBox({
+					type: 'error',
+					buttons: ['Copy to clipboard', 'Ok'],
```
Comment (sindresorhus) [inline index.js]: `Ok` => `OK`

`Copy to clipboard` => `Copy Error` on macOS and `Copy error` on Windows/Linux.

File: index.js [hunk 3]
```diff
@@ -17,18 +19,35 @@ module.exports = options => {
...
+					defaultId: 1,
+					title,
+					noLink: true,
+					message: cleanStack(stack)
```
Comment (sindresorhus) [inline index.js]: Lets only put the actual error message here, and put the actual stack in `detail`.

You can use https://github.com/sindresorhus/extract-stack to get the stack without the error message.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "electron-unhandled",
-  "version": "0.1.1",
+  "version": "0.1.2",
```
Comment (sindresorhus) [inline package.json]: Don't bump this. I will do it when I'm ready to do a new version.

**Comment by sindresorhus** [comment]
Great! Thanks for working on this :)

File: index.js [hunk 5]
```diff
@@ -33,18 +35,18 @@ module.exports = options => {
 			const stack = err ? (err.stack || err.message || err || '[No error message]') : '[Undefined error]';
 
 			if (isReady) {
-				dialog.showMessageBox({
+				const btnIndex = dialog.showMessageBox({
 					type: 'error',
-					buttons: ['Copy to clipboard', 'Ok'],
+					buttons: [os.type === 'Darwin' ? 'Copy Error' : 'Copy error', 'OK'],
```
Comment (sindresorhus) [inline index.js]: `process.platform === 'darwin'` is the preferred way to check for macOS.

File: index.js [hunk 6]
```diff
@@ -17,18 +21,35 @@ module.exports = options => {
...
-			showErrorBox(title, cleanStack(stack));
+
+			if (isReady) {
+				const btnIndex = dialog.showMessageBox({
+					type: 'error',
+					buttons: [process.platform === 'darwin' ? 'Copy Error' : 'Copy error', 'OK'],
+					defaultId: 1,
+					title,
+					noLink: true,
+					message: err.message,
```
Comment (sindresorhus) [inline index.js]: `err.message` is not guaranteed to be an Error object. While not recommended, people can reject a promise with any value.

Comment (armaldio) [inline index.js]: What to do then ?

Comment (sindresorhus) [inline index.js]: Same as done with `stack` on line 34. You have to account for all the cases...

Comment (armaldio) [inline index.js]: Like this ?

`const message = err ? (err.message || err.stack || err || '[No error message]') : '[Undefined error]';`

Comment (sindresorhus) [inline index.js]: Yes, except for `err.stack`.

**Comment by sindresorhus** [comment]
Thanks :)

**Comment by armaldio** [comment]
Np :)
Maybe you should close the issue https://github.com/sindresorhus/electron-unhandled/issues/2 too ?

**Comment by sindresorhus** [comment]
New version is out :)

**Comment by armaldio** [comment]
Yay!
Tahnk you very much :)

**Pull request merged**